File: app/src/main/java/com/example/volleyrvpractice/Network/GlideManager.java
package com.example.volleyrvpractice.Network;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.request.target.CustomTarget;
import com.bumptech.glide.request.transition.Transition;
import com.example.volleyrvpractice.Recipe.RecipeModel;

import java.util.List;

public class GlideManager {
    public static CustomTarget<Bitmap> loadImage(Context ct, String url, final int i, final GlideCallbackListener listener) {
        return Glide.with(ct).asBitmap().load(url).diskCacheStrategy(DiskCacheStrategy.NONE).into(new CustomTarget<Bitmap>() {
            @Override
            public void onResourceReady(@NonNull Bitmap resource, @Nullable Transition<? super Bitmap> transition) {
                //Log.d("onResourceReady",resource.toString();
                listener.getBitmap(resource, i);
            }

            @Override
            public void onLoadCleared(@Nullable Drawable placeholder) {

            }
        });
    }
}

File: app/src/main/java/com/jasonstudio/cookbook2/Network/GlideManager.kt
package com.example.volleyrvpractice.Network

import android.content.Context
import com.bumptech.glide.request.target.CustomTarget
import android.graphics.Bitmap
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import android.graphics.drawable.Drawable
import com.example.volleyrvpractice.R
import com.android.volley.toolbox.Volley
import com.example.volleyrvpractice.Network.CallbackListener
import com.android.volley.toolbox.JsonObjectRequest
import org.json.JSONObject
import org.json.JSONException
import com.android.volley.VolleyError
import com.bumptech.glide.request.transition.Transition
import kotlin.Throws

object GlideManager {
    fun loadImage(
        ct: Context?,
        url: String?,
        i: Int,
        listener: GlideCallbackListener
    ): CustomTarget<Bitmap> {
        return Glide.with(ct!!).asBitmap().load(url).diskCacheStrategy(DiskCacheStrategy.NONE)
            .into(object : CustomTarget<Bitmap>(556, 370) {
                override fun onResourceReady(resource: Bitmap, transition: Transition<in Bitmap>?) {
                    //Log.d("onResourceReady",resource.toString();
                    listener.getBitmap(resource, i)
                }

                override fun onLoadCleared(placeholder: Drawable?) {}
            })
    }
}